week1/homework/src/server.js

```diff
@@ -10,6 +10,30 @@ function createServer(port) {
 
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
+    function sendResponse(status, stateValue, contentType) {
```

Comment:
Try not to define functions inside of other functions. `sendResponse` here gets defined every time we call `http.createServer`.

Reply:
Thanks I corrected that. and this is to make performance better and not keep the stack occupied. is this right?

Reply:
Yes, it is about performance. Defining a function like this is the same as assigning any kind of value to a variable. If you do it inside a function, the JS engine will have to recreate that function (or object, array, etc.) each time it is called. Without going into the details, you pay that with memory and CPU usage. If you're interested, this is a good article: https://www.reaktor.com/blog/javascript-performance-fundamentals-make-bluebird-fast/